### PR TITLE
Support insertion in SyntaxRewriter

### PR DESCRIPTION
**Pull request by Veykril**

### CODE REVIEW

File: crates/syntax/src/algo.rs [hunk 1]
```diff
@@ -289,11 +289,19 @@ fn _replace_children(
     with_children(parent, new_children)
 }
 
+#[derive(Debug, PartialEq, Eq, Hash)]
+enum InsertPos {
+    FirstChildOf(SyntaxNode),
+    Before(SyntaxElement),
+    After(SyntaxElement),
```
Comment (matklad) [inline crates/syntax/src/algo.rs:296]: For public insert position, we need to suport both before and after for nicer API. But I wonder if this internal type can get by only with `After`?

Comment (Veykril) [reply crates/syntax/src/algo.rs:296]: If we can figure out whether something is the first child of a node or not we could I imagine. But I don't know how I could possibly check that.

Comment (matklad) [reply crates/syntax/src/algo.rs:296]: `it.prev_sibling().is_none()`

Comment (Veykril) [reply crates/syntax/src/algo.rs]: Okay I managed to remove it

**Pull request merged**